src/components/Watch/Player/Player.tsx

```diff
@@ -0,0 +1,123 @@
+"use client";
+
+import FullScreen from "../FullScreen/FullScreen";
+import PlayPause from "../PlayPause/PlayPause";
+import Skip from "../Skip/Skip";
+import Volume from "../Volume/Volume";
+import TimeBar from "../TimeBar/TimeBar";
+
+import "./Player.css";
+import React, { useState, useRef, useEffect } from "react";
+
+interface PlayerProps {
+  src: string;
+}
+
+function Player({ src }: PlayerProps) {
+  const videoRef = useRef<HTMLVideoElement | null>(null);
+  const [isPlaying, setIsPlaying] = useState(false);
+  const [currentTime, setCurrentTime] = useState(0);
+  const [duration, setDuration] = useState(0);
+  const [volume, setVolume] = useState(1);
+  const [isMuted, setIsMuted] = useState(false);
+  const [isFullScreen, setIsFullScreen] = useState(false);
+
+  useEffect(() => {
+    if (videoRef.current) {
+      // Update currentTime and duration when the video can play
+      videoRef.current.addEventListener("canplay", () => {
+        setDuration(videoRef.current.duration);
+      });
+
+      // Update currentTime while the video is playing
+      videoRef.current.addEventListener("timeupdate", () => {
+        setCurrentTime(videoRef.current.currentTime);
+      });
+
+      // Detect full-screen change
+      const handleFullScreenChange = () => {
+        setIsFullScreen(
+          !!(
+            document.fullscreenElement ||
+            document.mozFullScreenElement ||
+            document.webkitFullscreenElement ||
+            document.msFullscreenElement
+          ),
+        );
+      };
```

Comment:
```suggestion
    if (videoRef.current) {
      // Update currentTime and duration when the video can play
      videoRef.current.addEventListener("canplay", () => {
        if(videoRef.current) setDuration(videoRef.current.duration);
      });

      // Update currentTime while the video is playing
      videoRef.current.addEventListener("timeupdate", () => {
        if(videoRef.current) setCurrentTime(videoRef.current.currentTime);
      });

      // Detect full-screen change
      const handleFullScreenChange = () => {
        setIsFullScreen(
          !!(
            document.fullscreenElement ||
            document.mozFullScreenElement ||
            document.webkitFullscreenElement ||
            document.msFullscreenElement
          ),
        );
      };
```

이렇게 null check를 addEventListener 내부의 콜백에서 한 번 더 해주어야 타입 오류가 나지 않습니다. 왜일까요? 바로 eventListener에 전달한 callback은 언제 어떤 상황에서 작동될 지 모르기 때문입니다! 즉 이벤트 리스너가 추가되는 useEffect 내에서는 26행의 if문으로 인해 videoRef.current의 존재가 보장되었지만, 막상 콜백이 작동하는 순간에는 videoRef.current가 존재하지 않을 것을 걱정하여 타입스크립트가 오류를 표시하는 것이지요.